### PR TITLE
{matrix-synapse-tools.matrix-synapse-diskspace-janitor,nixos/matrix-synapse-diskspace-janitor}: init

### PR DESCRIPTION
**Pull request by ckiee**
(2023-06-29: *This may now be redundant, as anecdotally disk space usage hasn't been as big of a problem for me and cyberia recently*)

###### Description of changes

I had no intention to work on this but the Matrix gods decided it'd be a great time to fill up my whole disk.. (albeit RELATIVELY slowly)

Might undraft one day but code reviews will be dismissed for now— anyone can take over if they want and finish this.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/misc/ids.nix [hunk 1]
```diff
@@ -356,6 +356,7 @@ in
       rstudio-server = 324;
       localtimed = 325;
       automatic-timezoned = 326;
+      synapse-diskspace-janitor = 400; # TODO: change to next sequential before merge!!!!!!!!!!!!!
```
Comment (rnhmjoj) [inline nixos/modules/misc/ids.nix:359]: I don't think this is needed: it doesn't look like this service manages a lot of state files, so you should use systemd `DynamicUser` mechanism and let systemd deals with uid/gids.

File: nixos/modules/services/matrix/synapse-diskspace-janitor.nix [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    systemd.services.matrix-synapse-diskspace-janitor = {
+      description = "Matrix Synapse Disk Space Janitor";
+      documentation = [ "https://git.cyberia.club/cyberia/matrix-synapse-diskspace-janitor" ];
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
+      script = ''
+        ${lib.optionalString (cfg.adminTokenFile != null) ''
+          # Environment variables override the config file.
+          export JANITOR_MATRIXADMINTOKEN=$(cat "''${CREDENTIALS_DIRECTORY}/admin-token-file")
```
Comment (rnhmjoj) [inline nixos/modules/services/matrix/synapse-diskspace-janitor.nix:69]: Using `systemd.services.matrix-synapse-diskspace-janitor.environment` would be cleaner.

File: nixos/modules/misc/ids.nix (hunk 1 above)
Comment (ckiee) [reply nixos/modules/misc/ids.nix:359]: Yes, I tried that, but systemd gave us obscure errors for the `start` task:

```
Feb 25 03:37:48 bokkusu systemd[1]: matrix-synapse-diskspace-janitor.service: Failed to run 'start' task: Invalid argument
Feb 25 03:37:48 bokkusu systemd[1]: matrix-synapse-diskspace-janitor.service: Failed with result 'resources'.
```

File: nixos/modules/services/matrix/synapse-diskspace-janitor.nix (hunk 2 above)
Comment (ckiee) [reply nixos/modules/services/matrix/synapse-diskspace-janitor.nix:69]: Will do next iterating-over-our-PRs iteration.

File: nixos/modules/misc/ids.nix (hunk 1 above)
Comment (SuperSandro2000) [reply nixos/modules/misc/ids.nix:359]: You can just create a user with users.users.

**Comment by ckiee** [comment]
Closing this due to inactivity and a lack of my interest.

**Pull request closed without merging**